### ISSUE TITLE
C# extension stopped working suddenly

### ISSUE THREAD

**Issue by achobanov** [dotnet/vscode-csharp#7734]
Type: <b>Bug</b>

## Issue Description ##
Opening a new issue as instructed here: https://github.com/dotnet/vscode-csharp/issues/7514#issuecomment-2347181280

Initially it seemed like only limited functionality (such as the quick fix actions) wasn't working, however it seems like its a lot deeper now:
- compile time analysis is non-existent
- types are no longer resolved

Some stuff is working - for example if I hoven on a field, it shows me that's a readonly field of a certain type, although this could be VS code being clever, as it only works for members of the current class. Short story shorter - it's not working as expected at all. 

## Logs ##
I've set log level to trace.

        
### C# log ###
[vscode-c#-log.txt](https://github.com/user-attachments/files/17611710/vscode-c.-log.txt)

### C# LSP Trace Logs ###
[vscode-c#-lsp-log.txt](https://github.com/user-attachments/files/17611711/vscode-c.-lsp-log.txt)

## Environment information ##

**VSCode version**: 1.93.1
**C# Extension**: 2.50.27
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:           8.0.403
 Commit:            c64aa40a71
 Workload version:  8.0.400-manifests.e0880c5d
 MSBuild version:   17.11.9+a69bbaaf5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.403\

.NET workloads installed:
Configured to use loose manifests when installing new manifests.
 [android]
   Installation Source: SDK 8.0.400, VS 17.10.35326.205, VS 17.11.35327.3
   Manifest Version:    34.0.143/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.android\34.0.143\WorkloadManifest.json
   Install Type:        FileBased

 [aspire]
   Installation Source: SDK 8.0.400, VS 17.10.35326.205, VS 17.11.35327.3
   Manifest Version:    8.2.2/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.2.2\WorkloadManifest.json
   Install Type:        FileBased

 [ios]
   Installation Source: SDK 8.0.400, VS 17.10.35326.205, VS 17.11.35327.3
   Manifest Version:    18.0.8303/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.ios\18.0.8303\WorkloadManifest.json
   Install Type:        FileBased

 [maccatalyst]
   Installation Source: SDK 8.0.400, VS 17.10.35326.205, VS 17.11.35327.3
   Manifest Version:    18.0.8303/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maccatalyst\18.0.8303\WorkloadManifest.json
   Install Type:        FileBased

 [maui-windows]
   Installation Source: SDK 8.0.400, VS 17.10.35326.205, VS 17.11.35327.3
   Manifest Version:    8.0.82/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maui\8.0.82\WorkloadManifest.json
   Install Type:        FileBased

Host:
  Version:      8.0.10
  Architecture: x64
  Commit:       81cabf2857

.NET SDKs installed:
  6.0.419 [C:\Program Files\dotnet\sdk]
  7.0.410 [C:\Program Files\dotnet\sdk]
  8.0.201 [C:\Program Files\dotnet\sdk]
  8.0.302 [C:\Program Files\dotnet\sdk]
  8.0.306 [C:\Program Files\dotnet\sdk]
  8.0.403 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.27 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|auto-close-tag|formulahendry|0.5.15|formulahendry.auto-close-tag-0.5.15|
|csdevkit|ms-dotnettools|1.11.14|ms-dotnettools.csdevkit-1.11.14-win32-x64|
|csharp|ms-dotnettools|2.50.27|ms-dotnettools.csharp-2.50.27-win32-x64|
|dotnet-maui|ms-dotnettools|1.4.36|ms-dotnettools.dotnet-maui-1.4.36-win32-x64|
|vscode-dotnet-runtime|ms-dotnettools|2.2.2|ms-dotnettools.vscode-dotnet-runtime-2.2.2|
|vscode-icons|vscode-icons-team|12.9.0|vscode-icons-team.vscode-icons-12.9.0|
|vscode-todo-highlight|wayou|1.0.5|wayou.vscode-todo-highlight-1.0.5|
|vscode-yaml|redhat|1.15.0|redhat.vscode-yaml-1.15.0|
|vscodeintellicode-csharp|ms-dotnettools|2.1.11|ms-dotnettools.vscodeintellicode-csharp-2.1.11-win32-x64|;

</details>

Extension version: 2.50.27
VS Code version: Cursor 0.42.4 (b1e87884330fc271d5eb589e368c35f14e76dec0, 2024-10-29T17:17:29.347Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz (6 x 2808)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.92GB (10.60GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

[extension-log-folder.zip](https://github.com/user-attachments/files/17611708/extension-log-folder.zip)

**Comment by achobanov** [comment]
Update: basic compile info works for items outside of the current file. I.e. functionality such as `GoToDefinition` now suddenly works. Quick fix actions stull fumble though.

**Comment by dibarbet** [comment]
@achobanov a couple questions
1.  What files are you seeing issues in - razor, or normal c#, or both?
2.  When you open a file that is having issues - what project context does the C# extension find the file in?  You can find that here - https://github.com/dotnet/vscode-csharp/blob/main/SUPPORT.md#project-loading-problems
3.  In the solution explorer, are you seeing everything loaded properly?  For example do you see all the expected packages and references showing up?
  ![Image](https://github.com/user-attachments/assets/85683d82-a0e8-40dd-b97e-8cd17d9dc284)
4.  Just want to confirm you're still on VSCode version 1.93.1 - in version 1.95.0 there is a bug where code actions are not provided if the server takes longer than 1.2 seconds.

A short gif or screencap of exactly what is going wrong would be super helpful.

**Comment by JohnLudlow** [comment]
Not sure if this is the same error but I'm getting very similar behaviour

These errors pop up after opening a folder with a C# project:

![Image](https://github.com/user-attachments/assets/35ef73c8-352d-4234-aa3c-f81ef78abb0f)

I'm only getting file / text-based completion, no semantic language-aware completion:

![Image](https://github.com/user-attachments/assets/fe24f18a-a33c-4180-b587-b22192df656d)

Log from Output / C#

```
Using dotnet configured on PATH
Dotnet path: C:\Program Files\dotnet\dotnet.exe
Activating C# + C# Dev Kit + C# IntelliCode...
waiting for named pipe information from server...
[stdout] {"pipeName":"\\\\.\\pipe\\a74c7130"}
received named pipe information from server
attempting to connect client to server...
client has connected to server
[Info  - 9:43:26 AM] [Program] Language server initialized
[Error - 9:43:26 AM] Client Microsoft.CodeAnalysis.LanguageServer: connection to server is erroring. Shutting down server.
[Error - 9:43:26 AM] Client Microsoft.CodeAnalysis.LanguageServer: connection to server is erroring. Shutting down server.
[Error - 9:43:26 AM] Stopping server failed
TypeError: Promise.race(...).then(...).finally is not a function
	at a.shutdown (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2111629)
	at a.stop (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2110680)
	at a.stop (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2273841)
	at a.handleConnectionError (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2117187)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Error - 9:43:26 AM] Stopping server failed
Error: Client is stopping but no stop promise available.
	at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (c:\Users\HAL04982\.vscode\extensions\qub.qub-xml-vscode-1.2.8\node_modules\zone.js\dist\zone-node.js:811:33)
	at new AppInsightsAsyncCorrelatedErrorWrapper (c:\Users\HAL04982\.vscode\extensions\qub.qub-xml-vscode-1.2.8\node_modules\applicationinsights\out\AutoCollection\CorrelationContextManager.js:172:18)
	at a.shutdown (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2110959)
	at a.stop (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2110680)
	at a.stop (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2273841)
	at a.handleConnectionError (c:\Users\HAL04982\.vscode\extensions\ms-dotnettools.csharp-2.50.27-win32-x64\dist\extension.js:2:2117187)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Error - 9:43:26 AM] Stopping server failed
  Message: This socket has been ended by the other party
  Code: -32099 
[Error - 9:43:26 AM] Connection to server got closed. Server will not be restarted.
[stderr] Unhandled exception: [stderr] System.Text.Json.JsonReaderException: 'C' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0.
   at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
   at System.Text.Json.Utf8JsonReader.ConsumeValue(Byte marker)
   at System.Text.Json.Utf8JsonReader.ReadFirstToken(Byte first)
   at System.Text.Json.Utf8JsonReader.ReadSingleSegment()
   at System.Text.Json.Utf8JsonReader.Read()
   at System.Text.Json.JsonDocument.Parse(ReadOnlySpan`1 utf8JsonSpan, JsonReaderOptions readerOptions, MetadataDb& database, StackRowStack& stack)
   at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory`1 utf8Json, JsonReaderOptions readerOptions, Byte[] extraRentedArrayPoolBytes, PooledByteBufferWriter extraPooledByteBufferWriter)
   at System.Text.Json.JsonDocument.Parse(ReadOnlySequence`1 utf8Json, JsonDocumentOptions options)
   at StreamJsonRpc.SystemTextJsonFormatter.Deserialize(ReadOnlySequence`1 contentBuffer, Encoding encoding)
   at StreamJsonRpc.PipeMessageHandler.DeserializeMessageAsync(Int32 contentLength, Encoding specificEncoding, Encoding defaultEncoding, CancellationToken cancellationToken)
   at StreamJsonRpc.HeaderDelimitedMessageHandler.ReadCoreAsync(CancellationToken cancellationToken)
   at StreamJsonRpc.MessageHandlerBase.ReadAsync(CancellationToken cancellationToken)
   at StreamJsonRpc.JsonRpc.ReadAndHandleRequestsAsync()
   at Microsoft.CodeAnalysis.LanguageServer.LanguageServer.LanguageServerHost.WaitForExitAsync() in /_/src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/LanguageServerHost.cs:line 69
   at Program.<<Main>$>g__RunAsync|0_0(ServerConfiguration serverConfiguration, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs:line 143
   at Program.<<Main>$>g__RunAsync|0_0(ServerConfiguration serverConfiguration, CancellationToken cancellationToken) in /_/src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs:line 151
   at System.CommandLine.CliCommand.<>c__DisplayClass32_0.<<SetAction>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Invocation.InvocationPipeline.InvokeAsync(ParseResult parseResult, CancellationToken cancellationToken)
Language server process exited with 1

```

**Comment by achobanov** [comment]
@dibarbet 
1. All kinds of files
2. The context seems correct - it's the name of the project it's part of
3. Dependencies look good at a glace:
 ![Image](https://github.com/user-attachments/assets/6870355a-11ce-4a49-b335-00306fe7db25)
4. To be specific I'm using Cursor IDE (VS Code fork, you might have heard of it), which is currently based on VS Code 1.93.1. However I am experiencing the same issues on a separate VS Code installation which is currently version 1.90.2. Both behave in the exact same way

Here's a video:https://github.com/user-attachments/assets/1b511a90-e423-45a8-8913-a009ecc0648f

**Comment by dibarbet** [comment]
@achobanov thank you for the video, that and the trace logs you originally attached helped a lot!

It appears to be an issue applying a C# code action when the code action returns edits inside a razor file.  Instead of getting the top level razor file, we return edits in the virtual C# file.  Note that this also applies to FAR/GTD when invoking from a C# file -> razor file.

```
"textDocument": {
    "version": null,
    "uri": "virtualcsharp-razor:///<somefile>.razor__virtual.cs"
},
"edits": [
    {
        "range": {
            "start": {
                "line": 575,
                "character": 14
            },
            "end": {
                "line": 575,
                "character": 16
            }
        },
        "newText": "P"
    }
]
```

**Comment by ryzngard** [comment]
Thank you for helping with so much interaction and traces. We're evaluating solutions so I can't give an ETA, but I can tell you this helps push priorities. Long term we have a solution for this in a big architecture change that we are calling "cohosting" (if you see anything about that in the razor repo) but short term I'm going to see if we can find a tactical fix. 

> I don't understand "Note that this also applies to FAR/GTD when invoking from a C# file -> razor file".

The general problem is how the razor compiler (which produces csharp files) communicates the content of those files to the csharp server across processes. You may be asking yourself why two processes? Well! That's what cohosting is going to solve :). Until then though the problem is that by design LSP messages that notify of content from an editor it's supposed to block other requests until that request is handled. Think of it like a user typing and the editor is informing of the changes because the file isn't saved to disk. You want everything to be up to date when asking for things like what semantic tokens are available, what completions should be happening, and what symbols are defined. The same problem applies with how we're communicating the generated files to the csharp server. So requests for FAR/GTD have to first know about the content before they can accurately determine the answer. If the communication of csharp files to the LSP server is blocking things it will slow down the system. 

Hopefully that's helpful. It's a bit of a late night rant from me to say: yes, we understand the problem. yes, we're working on it. I can't give an ETA but solving this is high on my to-do list.

**Comment by achobanov** [comment]
@ryzngard Thanks for the clarification. No worries I understand the situation. It's great to hear there is a fundamental solution planned. In the meantime I'll assume I can't rely on VS Code to fix stuff in razor files. However I've encountered other (on the surface similar issues) that are VS Code or razor related. Can I ask for directions on how and where to report those:
- VS Code  namespace code fix - Auto-fixing the namespace to match the directory structure doesn't seem to work at all, not only for razor files;
- Razor - when a namespace for a base component or other dependency (.razor file) is automatically updated by moving a file in Visual Studio's solution explorer it results in scrambled `@usings` in those razor files. Instead of renaming the used namespaces it inserts tabs everywhere and breaks the compilation
- Razor - Custom components are often flagged as not imported, while in fact they seem to be. This happens both in VS Code and VS and sometimes makes sense, for example when a used component has a compile-time error and thus it's .cs files aren't built, but more often than not it's a false alarm that lights up the entire file/project. I suspect global usings are casing this and am currently in the process of removing those and seeing if the issue persists
- VS Code's c# solution doesn't handle file renames well - sometimes it creates duplicate files. Furthermore it doens't seem to have an option to rename the class and file name at once
- VS Code's c# solution isn't very elegant for keyboard navigation - I cannot get it to create a new directory using a hotkey the same way I can do with VS's solution explorer (although there seem to be some keybinds for that, I cannot get them to work) and I often have to resort to using either VS or VS Code's File explorer instead.

This is an initial feedback of trying to use VS Code as aa main c# editor for about 10 days now. I'd appreciate your feedback

**Comment by ryzngard** [comment]
> VS Code namespace code fix - Auto-fixing the namespace to match the directory structure doesn't seem to work at all, not only for razor files;

There's a few nuances to that. Do you mean the code action doesn't apply to C# files as well? We have a known issue for it not updating razor files https://github.com/dotnet/razor/issues/6167 

> Razor - when a namespace for a base component or other dependency (.razor file) is automatically updated by moving a file in Visual Studio's solution explorer it results in scrambled @usings in those razor files. Instead of renaming the used namespaces it inserts tabs everywhere and breaks the compilation

This actually was just recently fixed by https://github.com/dotnet/razor/pull/11146 and will ship in VS 17.13 P2 

> Razor - Custom components are often flagged as not imported, while in fact they seem to be. This happens both in VS Code and VS and sometimes makes sense, for example when a used component has a compile-time error and thus it's .cs files aren't built, but more often than not it's a false alarm that lights up the entire file/project. I suspect global usings are casing this and am currently in the process of removing those and seeing if the issue persists

Is this RZ10012 (I hope I got the number right) saying a component isn't found? 

> VS Code's c# solution doesn't handle file renames well - sometimes it creates duplicate files. Furthermore it doens't seem to have an option to rename the class and file name at once

If it's the solution explorer can you file a separate bug? I don't know about the rename dialog being customized like it is in VS. I'm assuming that's a VS Code limitation because they own that UI and we implement the LSP contract.

> VS Code's c# solution isn't very elegant for keyboard navigation - I cannot get it to create a new directory using a hotkey the same way I can do with VS's solution explorer (although there seem to be some keybinds for that, I cannot get them to work) and I often have to resort to using either VS or VS Code's File explorer instead.

Another issue to track this would be great :)

**Comment by ryzngard** [comment]
I also realized I was combining https://github.com/dotnet/vscode-csharp/issues/7582 with my response last night. It all kind of folds into the same problem space in my head but I realize they're two separate issues.

**Comment by achobanov** [comment]
> There's a few nuances to that. Do you mean the code action doesn't apply to C# files as well? We have a known issue for it not updating razor files https://github.com/dotnet/razor/issues/6167

I certainly thought so, but I cannot replicate it now, so I guess it was either temporary or I got it wrong, so scratch that one.

> This actually was just recently fixed by https://github.com/dotnet/razor/pull/11146 and will ship in VS 17.13 P2

FML it cost me half a day of manual usings editing in my razor files the other day.. Anyway happy it's shipped. 

>Is this RZ10012 (I hope I got the number right) saying a component isn't found?

Yes thats the one: _Found markup element with unexpected name 'WithdrawForm'. If this is intended to be a component, add a @using directive for its namespace.Razor(RZ10012)_

> If it's the solution explorer can you file a separate bug? I don't know about the rename dialog being customized like it is in VS. I'm assuming that's a VS Code limitation because they own that UI and we implement the LSP contract.

Just to clarify - it's not that I need to customize the visuals of the rename dialog. I think you can easily do it with an option in the extension, which when ON will rename the file along with the type name. 

> Another issue to track this would be great :)

I'll need some time to setup the keybinds in order to report exactly what's going wrong with details.

**Comment by ryzngard** [comment]
> Yes thats the one: Found markup element with unexpected name 'WithdrawForm'. If this is intended to be a component, add a @using directive for its namespace.Razor(RZ10012)

We definitely have had issues about this in the past and work on fixes. If you hit it again report another issue so we can get logs. Good information to include also is: Does it go away? Does it happen after it previously was working? Are they from a component in the same project, different project, or from a nuget package? 

> Just to clarify - it's not that I need to customize the visuals of the rename dialog. I think you can easily do it with an option in the extension, which when ON will rename the file along with the type name.

That's not a bad idea. Can you file an issue for this and I'll mark as a feature request? That way it can go onto the queue and be voted on etc so we know how to prioritize.

> One further question - can you briefly explain or direct me to a good read on the topic if you know a good resource (I couldnt find one) regarding .razor roslyn analyzers? I'd like to enforce some code style and formatting rules, but standard analyzers don't seem to work. I've opened up a [question here](https://stackoverflow.com/questions/79172694/how-to-create-codeanalysis-analyzer-for-razor-files)

This isn't currently a scenario we've had a lot of requests for. Roslyn analyzers aren't intended to work because razor files are `AdditionalDocuments` and not something parsed by Roslyn. You can enforce styling on the generated cs but it likely would be hard to do since you're depending on generation behavior that may change over time. I recently did a hackathon style project to find out what it would take to have a support system for this and it was.... a lot. It's never too little to get something on the books and your +1 so we know what the need for it is. In the meantime you could probably use a linter extension in VS Code. I'm less familiar with that so I don't have an immediate answer but it would unblock VS Code users and may have linter support for CI builds.

**Comment by achobanov** [comment]
> If you hit it again report another issue so we can get logs.
It happens every time I open the solution for a lot of components (maybe not all but quite a lot) and then it resolves itself. I'm guessing that's part of the loading process and doesn't warrant an issue? I'll report report the next sticky occurrence I get.

**Comment by ryzngard** [comment]
If it's only at startup we're working on improving that. Tracking with https://github.com/dotnet/vscode-csharp/issues/7582  and https://github.com/dotnet/razor/issues/10600 . If it's sticky that's definitely not good and when logs would help.

**Comment by achobanov** [comment]
> If it's sticky that's definitely not good and when logs would help
Here's a case that's also happening often, It's not a big deal but you need to rm bin/obj and restart the ide in order to fix: https://github.com/dotnet/vscode-csharp/issues/7798

> Another issue to track this would be great :)
And here's the solution navigation suggestions: https://github.com/dotnet/vscode-csharp/issues/7799
